lib/virtmanager.pm

```diff
@@ -584,4 +584,27 @@ sub create_guest {
     assert_and_click 'virtman_guest_begin_install';
 }
 
+# Expecting to see the guest screen
+sub detect_login_screen {
+    my $timeout = shift // 5;
+    if (!check_screen 'virt-manager_login-screen', $timeout) {
```

Comment:
Bad case example:
&nbsp;&nbsp; `$timeout` param is set to 3 minutes
&nbsp;&nbsp; `virt-manager_login-screen` is visible after 30s
You will wait 3 minutes until first `if` times out, then 2 x 5s for second and third`if` (which is not needed cause you already have login screen). It's usually better to assert `something`, not `everything else`.

You can for example:
```perl
    my $timeout = shift // 5;

    return if check_screen('virt-manager_login-screen', $timeout);
    wait_still_screen 3;    # Connecting to guest's console
    mouse_set(30, 200);     # Go inside of the guest's console
    mouse_click();
    save_screenshot();
    hold_key "ctrl-alt";       # Escape from the guest's console
    release_key "ctrl-alt";    # Now the mouse pointer is free
    mouse_set(300, 70);

    return if check_screen('virt-manager_login-screen', 5);
    send_key 'esc';
    send_key 'backspace';

    return if check_screen('virt-manager_login-screen', 5);
    assert_and_click 'virt-manager_send-key';
    assert_and_click 'virt-manager_ctrl-alt-f2';

    assert_screen "virt-manager_login-screen";
}
```

Reply:
That really looks better, @kravciak, I just need to probably test it which will take me few hours perhaps I've no hypervisor installed right now.